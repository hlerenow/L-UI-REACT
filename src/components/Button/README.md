# Button 文档

asdasd

|  表头   | 表头  |
|  ----  | ----  |
| 单元格  | 单元格 |
| 单元格  | 单元格 |

```javascript
<Button type={bType}>{bType}</Button>
```
