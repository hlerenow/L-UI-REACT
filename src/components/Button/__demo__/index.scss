.item-box {
  display: flex;
  .item {
    padding: 5px;
  }
}