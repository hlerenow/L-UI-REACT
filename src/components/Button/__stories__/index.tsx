/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-unused-vars */
import * as React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import './index.scss';
import { Button, BaseButtonProps } from '../index';

export default {
  title: 'L-UI/Base/Button',
  component: Button,
  argTypes: {
  },
} as Meta;

const Template: Story<BaseButtonProps> = (args) => (
  <Button {...args}>{args.type}</Button>
);

export const Default = Template.bind({});

Default.args = {
  type: 'default',
};

export const Primary = Template.bind({});
Primary.args = {
  type: 'primary',
};

export const Warn = Template.bind({});

Warn.args = {
  type: 'warn',
};

export const Danger = Template.bind({});

Danger.args = {
  type: 'danger',
};

export const Text = Template.bind({});

Text.args = {
  type: 'text',
};

export const Info = Template.bind({});

Info.args = {
  type: 'info',
};
