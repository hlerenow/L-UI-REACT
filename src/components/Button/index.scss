@import '../../styles/variables.scss';

.l-button {
  box-sizing: border-box;
  padding: $ButtonPaddingY  $ButtonPaddingX;
  border: $borderSize solid $borderColor;
  border-radius: $borderRadius;
  background-color: white;
  cursor: pointer;
  outline: none;
  font-size: 14px;
  line-height: 1;

  &.primary {
    border: none;
    background-color: $primaryColor;
    color: white;
  }
}
