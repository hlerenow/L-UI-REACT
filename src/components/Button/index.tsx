import * as React from 'react';

const Button = function Button() {
  return <div>12</div>;
};
export default Button;
