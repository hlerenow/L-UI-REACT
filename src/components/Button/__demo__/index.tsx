import * as React from 'react';
// eslint-disable-next-line no-unused-vars
import { Button, ButtonType, ButtonSize } from '@/src/components/Button/index';
import './index.scss';

type StateType = {
  refs: any,
  types?: string[]
}

class ButtonDemo extends React.Component<any, StateType> {
  constructor(props: any) {
    super(props);
    this.state = {
      refs: React.createRef(),
      types: ['default', 'primary', 'success', 'info', 'warn', 'danger'],
    };
  }

  onClick = () => {
    alert('被点击');
  }

  render() {
    const { onClick } = this;
    const { refs, types } = this.state;
    const sizeTypes: ButtonSize[] = ['medium', 'small', 'mini', 'small'];
    return (
      <div>
        <h2>Button</h2>
        <div className="item-box">
          {
            types.map((type: ButtonType) => (<div className="item" key={type} onClick={onClick} ref={refs}><Button type={type}>{type}</Button></div>))
          }
        </div>
        {/* plain */}
        <div className="item-box">
          {
            types.map((type: ButtonType) => (<div className="item" key={type} onClick={onClick} ref={refs}><Button plain type={type}>{type}</Button></div>))
          }
        </div>
        {/* round */}
        <div className="item-box">
          {
            types.map((type: ButtonType) => (<div className="item" key={type} onClick={onClick} ref={refs}><Button round type={type}>{type}</Button></div>))
          }
        </div>
        {/* round and plain */}
        <div className="item-box">
          {
            types.map((type: ButtonType) => (<div className="item" key={type} onClick={onClick} ref={refs}><Button round plain type={type}>{type}</Button></div>))
          }
        </div>
        {/* size */}
        <div className="item-box">
          {
            types.map((type: ButtonType, index) => (<div className="item" key={type} onClick={onClick} ref={refs}><Button size={sizeTypes[index]} type={type}>{type}</Button></div>))
          }
        </div>
      </div>
    );
  }
}

export default ButtonDemo;
