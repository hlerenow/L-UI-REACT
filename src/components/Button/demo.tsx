import * as React from 'react';
import Button from './index';

type StateType = {
  types: string[]
}

class ButtonDemo extends React.Component<any, StateType> {
  constructor(props: any) {
    super(props);
    this.state = {
      types: ['default', 'primary', 'success', 'info', 'warning', 'danger'],
    };
  }

  render() {
    const { types } = this.state;
    return (
      <div>
        <h2>Button</h2>
        {
          types.map((type) => (<div style={{ padding: '5px', float: 'left' }} key={type}><Button type={type}>{type}</Button></div>))
        }
      </div>
    );
  }
}

export default ButtonDemo;
