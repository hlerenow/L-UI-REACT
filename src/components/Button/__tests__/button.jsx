import * as React from 'react';
import { shallow } from 'enzyme';
import Button from '@/src/components/Button/index';
import imageTest from '@/tests/imageSnapshots/imageTest';

describe('Test button', () => {
  it('render a Button', () => {
    const button = shallow(<Button>Default</Button>);
    expect(button.html()).toMatchSnapshot();
    expect(button.text()).toEqual('Default');
    expect(button.hasClass('l-button--default')).toBe(true);
  });

  it('button type primary', () => {
    const button = shallow(<Button type="primary">primary</Button>);
    expect(button.html()).toMatchSnapshot();
    expect(button.text()).toEqual('primary');
    expect(button.hasClass('l-button--primary')).toBe(true);
  });
});

describe('Button image', () => {
  imageTest((
    <>
      <Button>Default</Button>
      <Button type="primary">Default</Button>
    </>
  ));
});
