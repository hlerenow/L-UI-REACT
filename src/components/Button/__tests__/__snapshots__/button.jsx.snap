// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button button type primary 1`] = `"<button type=\\"button\\" class=\\"l-button l-button--primary\\">primary</button>"`;

exports[`Test button render a Button 1`] = `"<button type=\\"button\\" class=\\"l-button l-button--default\\">Default</button>"`;
